app/assets/scripts/map.js

```diff
@@ -218,34 +244,27 @@ function populateListDiv() {
   var markersToShow = markers;
   $("#list").empty();
 
-  var list = document.getElementById('list');
+  var listResources = [];
   $.each(markersToShow, function(i, markerToShow) {
-    var listElement = document.createElement('a');
-    listElement.setAttribute('class', 'item');
-
-    var rightArrowElement = document.createElement('div');
-    rightArrowElement.setAttribute('class', 'right floated content');
-
-    var rightArrowImage = document.createElement('i');
-    rightArrowImage.setAttribute('class', 'right chevron icon');
-
-    rightArrowElement.appendChild(rightArrowImage);
-    listElement.appendChild(rightArrowElement);
-
-    var mainContent = document.createElement('div');
-    mainContent.setAttribute('class', 'content');
-
-    var placeTitle = document.createElement('div');
-    placeTitle.setAttribute('class', 'list-item-title');
-    $(placeTitle).html(markerToShow.title);
-
-    mainContent.appendChild(placeTitle);
-    listElement.appendChild(mainContent);
+    var listResource = {
+      name: markerToShow.title,
+      address: markerToShow.address,
+    };
+    listResources.push(listResource);
+  });
 
-    list.appendChild(listElement);
+  var listTemplate = $("#listview-template").html();
+  var compiledListTemplate = Handlebars.compile(listTemplate);
+  var context = {
+    resource: listResources,
+  };
+  var listView = compiledListTemplate(context);
+  $("#list").html(listView);
 
-    listElement.addEventListener('click', function() {
-      google.maps.event.trigger(markerToShow, 'click');
+  // Can only add handlers to elements in template after compilation
```

Comment:
what do you mean? when a new search/filter is done to change the list or when the list is initially rendered? it would be the same in both cases since a new search just reruns this method which creates the UI outlined by the `#listview-template`